(Documenting for myself on Monday) This is currently failing with

```
   tests/models/test_taskinstance.py:2288: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  airflow/utils/session.py:71: in wrapper
      return func(*args, session=session, **kwargs)
  airflow/models/taskinstance.py:1621: in run
      session=session,
  airflow/utils/session.py:68: in wrapper
      return func(*args, **kwargs)
  airflow/models/taskinstance.py:1194: in check_and_change_state_before_execution
      dep_context=non_requeueable_dep_context, session=session, verbose=True
  airflow/utils/session.py:68: in wrapper
      return func(*args, **kwargs)
  airflow/models/taskinstance.py:1031: in are_dependencies_met
      for dep_status in self.get_failed_dep_statuses(dep_context=dep_context, session=session):
  airflow/models/taskinstance.py:1052: in get_failed_dep_statuses
      for dep_status in dep.get_dep_statuses(self, session, dep_context):
  airflow/ti_deps/deps/base_ti_dep.py:95: in get_dep_statuses
      yield from self._get_dep_statuses(ti, session, dep_context)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = <TIDep(Previous Dagrun State)>
  ti = <[DetachedInstanceError('Instance <TaskInstance at 0x7fefa43f9850> is not bound to a Session; attribute refresh operation cannot proceed') raised in repr()] TaskInstance object at 0x7fefa43f9850>
  session = <sqlalchemy.orm.session.Session object at 0x7fefd5d5f050>
  dep_context = <airflow.ti_deps.dep_context.DepContext object at 0x7fefa4430050>
  
      @provide_session
      def _get_dep_statuses(self, ti, session, dep_context):
          if dep_context.ignore_depends_on_past:
              reason = "The context specified that the state of past DAGs could be ignored."
              yield self._passing_status(reason=reason)
              return
      
  >       if not ti.task.depends_on_past:
  E       AttributeError: 'MappedOperator' object has no attribute 'depends_on_past'
  
  airflow/ti_deps/deps/prev_dagrun_dep.py:41: AttributeError
```

--- a/airflow/models/base.py
+++ b/airflow/models/base.py
@@ -17,12 +17,19 @@
 # under the License.
 
 import functools
-from typing import Any, Type
+import logging
+from typing import TYPE_CHECKING, Any, Collection, Optional, Set, Type
 
+import jinja2
 from sqlalchemy import MetaData, String
 from sqlalchemy.ext.declarative import declarative_base
 
 from airflow.configuration import conf
+from airflow.templates import SandboxedEnvironment
+from airflow.utils.weight_rule import WeightRule
+
+if TYPE_CHECKING:
+    from airflow.models.dag import DAG
 
 SQL_ALCHEMY_SCHEMA = conf.get("core", "SQL_ALCHEMY_SCHEMA")
 
@@ -34,9 +41,123 @@ Base = declarative_base(metadata=metadata)  # type: Any
 ID_LEN = 250
 
 
-# used for typing
 class Operator:
-    """Class just used for Typing"""
+    """Common interface for operators, including unmapped and mapped."""
+
+    log: logging.Logger
+
+    upstream_task_ids: Set[str]
+    downstream_task_ids: Set[str]
+    weight_rule: str
+    priority_weight: int
+
+    # For derived classes to define which fields will get jinjaified.
+    template_fields: Collection[str]
+    # Defines which files extensions to look for in the templated fields.
+    template_ext: Collection[str]
+
+    def get_dag(self) -> "Optional[DAG]":
+        raise NotImplementedError()
+
+    @property
+    def dag_id(self) -> str:
+        """Returns dag id if it has one or an adhoc + owner"""
+        dag = self.get_dag()
+        if dag:
+            return self.dag.dag_id
+        return f"adhoc_{self.owner}"
+
+    def get_template_env(self) -> jinja2.Environment:
+        """Fetch a Jinja template environment from the DAG or instantiate empty environment if no DAG."""
+        dag = self.get_dag()
+        if dag:
+            return dag.get_template_env()
+        return SandboxedEnvironment(cache_size=0)
+
+    def prepare_template(self) -> None:
+        """Hook triggered after the templated fields get replaced by their content.
+
+        If you need your operator to alter the content of the file before the
+        template is rendered, it should override this method to do so.
+        """
+
+    def resolve_template_files(self) -> None:
+        """Getting the content of files for template_field / template_ext."""
+        if self.template_ext:
+            for field in self.template_fields:
+                content = getattr(self, field, None)
+                if content is None:
+                    continue
+                elif isinstance(content, str) and any(content.endswith(ext) for ext in self.template_ext):
+                    env = self.get_template_env()
+                    try:
+                        setattr(self, field, env.loader.get_source(env, content)[0])  # type: ignore
+                    except Exception:
+                        self.log.exception("Failed to resolve template field %r", field)
+                elif isinstance(content, list):
+                    env = self.get_template_env()
+                    for i, item in enumerate(content):
+                        if isinstance(item, str) and any(item.endswith(ext) for ext in self.template_ext):
+                            try:
+                                content[i] = env.loader.get_source(env, item)[0]  # type: ignore
+                            except Exception as e:
+                                self.log.exception(e)
+        self.prepare_template()
+
+    def get_direct_relative_ids(self, upstream: bool = False) -> Set[str]:
+        """Get direct relative IDs to the current task, upstream or downstream."""
+        if upstream:
+            return self.upstream_task_ids
+        return self.downstream_task_ids
+
+    def get_flat_relative_ids(
+        self,
+        upstream: bool = False,
+        found_descendants: Optional[Set[str]] = None,
+    ) -> Set[str]:
+        """Get a flat set of relative IDs, upstream or downstream."""
+        dag = self.get_dag()
+        if not dag:
+            return set()
+
+        if not found_descendants:
+            found_descendants = set()
+        relative_ids = self.get_direct_relative_ids(upstream)
+
+        for relative_id in relative_ids:
+            if relative_id not in found_descendants:
+                found_descendants.add(relative_id)
+                relative_task = dag.task_dict[relative_id]
+                relative_task.get_flat_relative_ids(upstream, found_descendants)
+
+        return found_descendants
+
+    @property
+    def priority_weight_total(self) -> int:
+        """
+        Total priority weight for the task. It might include all upstream or downstream tasks.
+
+        Depending on the weight rule:
+
+        - WeightRule.ABSOLUTE - only own weight
+        - WeightRule.DOWNSTREAM - adds priority weight of all downstream tasks
+        - WeightRule.UPSTREAM - adds priority weight of all upstream tasks
+        """
+        if self.weight_rule == WeightRule.ABSOLUTE:
+            return self.priority_weight
+        elif self.weight_rule == WeightRule.DOWNSTREAM:
+            upstream = False
+        elif self.weight_rule == WeightRule.UPSTREAM:
+            upstream = True
+        else:
+            upstream = False
+        dag = self.get_dag()
+        if dag is None:
+            return self.priority_weight
+        return self.priority_weight + sum(
+            dag.task_dict[task_id].priority_weight
+            for task_id in self.get_flat_relative_ids(upstream=upstream)
+        )
 
 
 def get_id_collation_args():

--- a/airflow/models/baseoperator.py
+++ b/airflow/models/baseoperator.py
@@ -1577,22 +1577,36 @@ class MappedOperator(Operator, LoggingMixin, DAGNode):
     subdag: None = attr.ib(init=False)
 
     @_is_dummy.default
-    def _is_dummy_default(self):
+    def _is_dummy_from_operator_class(self):
         from airflow.operators.dummy import DummyOperator
 
         return issubclass(self.operator_class, DummyOperator)
 
     @deps.default
-    def _deps_from_class(self):
+    def _deps_from_operator_class(self):
         return self.operator_class.deps
 
     @template_fields.default
-    def _template_fields_from_class(self):
+    def _template_fields_from_operator_class(self):
         return self.operator_class.template_fields
 
     @template_ext.default
-    def _template_ext_from_class(self):
+    def _template_ext_from_operator_class(self):
         return self.operator_class.template_ext
+
+    @task_type.default
+    def _task_type_from_operator_class(self):
+        # Can be a string if we are de-serialized
+        val = self.operator_class
+        if isinstance(val, str):
+            return val.rsplit('.', 1)[-1]
+        return val.__name__
+
+    @task_group.default
+    def _task_group_default(self):
+        from airflow.utils.task_group import TaskGroupContext
+
+        return TaskGroupContext.get_current_task_group(self.dag)
 
     @weight_rule.default
     def _weight_rule_from_kwargs(self) -> str:
@@ -1673,20 +1687,6 @@ class MappedOperator(Operator, LoggingMixin, DAGNode):
         for arg in self.mapped_kwargs.values():
             XComArg.apply_upstream_relationship(self, arg)
 
-    @task_type.default
-    def _default_task_type(self):
-        # Can be a string if we are de-serialized
-        val = self.operator_class
-        if isinstance(val, str):
-            return val.rsplit('.', 1)[-1]
-        return val.__name__
-
-    @task_group.default
-    def _default_task_group(self):
-        from airflow.utils.task_group import TaskGroupContext
-
-        return TaskGroupContext.get_current_task_group(self.dag)
-
     def get_dag(self) -> "Optional[DAG]":
         return self.dag
 
@@ -1751,6 +1751,14 @@ class MappedOperator(Operator, LoggingMixin, DAGNode):
                     'operator_extra_links',
                     'upstream_task_ids',
                     'task_type',
+                    # These are automatically populated from partial_kwargs. In
+                    # a perfect world, they should be properties like other
+                    # partial_kwargs-populated values e.g. 'queue' below, but we
+                    # must match BaseOperator's implementation and declare them
+                    # as writable attributes instead.
+                    'weight_rule',
+                    'priority_weight',
+                    'trigger_rule',
                 }
                 | {'template_fields'}
             )

--- a/airflow/models/baseoperator.py
+++ b/airflow/models/baseoperator.py
@@ -31,6 +31,7 @@ from typing import (
     Any,
     Callable,
     ClassVar,
+    Collection,
     Dict,
     FrozenSet,
     Iterable,
@@ -52,7 +53,6 @@ from dateutil.relativedelta import relativedelta
 from sqlalchemy.orm import Session
 from sqlalchemy.orm.exc import NoResultFound
 
-import airflow.templates
 from airflow.compat.functools import cached_property
 from airflow.configuration import conf
 from airflow.exceptions import AirflowException, TaskDeferred
@@ -249,6 +249,11 @@ class BaseOperatorMeta(abc.ABCMeta):
         )
 
 
+DEFAULT_QUEUE = conf.get("operators", "default_queue")
+DEFAULT_RETRIES = conf.getint("core", "default_task_retries", fallback=0)
+DEFAULT_WEIGHT_RULE = conf.get("core", "default_task_weight_rule", fallback=WeightRule.DOWNSTREAM)
+
+
 @functools.total_ordering
 class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
     """
@@ -428,11 +433,10 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
         that is visible in Task Instance details View in the Webserver
     """
 
-    # For derived classes to define which fields will get jinjaified
-    template_fields: Sequence[str] = ()
-    # Defines which files extensions to look for in the templated fields
-    template_ext: Sequence[str] = ()
-    # Template field renderers indicating type of the field, for example sql, json, bash
+    # Implementing Operator.
+    template_fields: Collection[str] = ()
+    template_ext: Collection[str] = ()
+
     template_fields_renderers: Dict[str, str] = {}
 
     # Defines the color in the UI
@@ -524,7 +528,7 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
         email: Optional[Union[str, Iterable[str]]] = None,
         email_on_retry: bool = conf.getboolean('email', 'default_email_on_retry', fallback=True),
         email_on_failure: bool = conf.getboolean('email', 'default_email_on_failure', fallback=True),
-        retries: Optional[int] = conf.getint('core', 'default_task_retries', fallback=0),
+        retries: Optional[int] = DEFAULT_RETRIES,
         retry_delay: Union[timedelta, float] = timedelta(seconds=300),
         retry_exponential_backoff: bool = False,
         max_retry_delay: Optional[Union[timedelta, float]] = None,
@@ -536,8 +540,8 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
         params: Optional[Dict] = None,
         default_args: Optional[Dict] = None,
         priority_weight: int = 1,
-        weight_rule: str = conf.get('core', 'default_task_weight_rule', fallback=WeightRule.DOWNSTREAM),
-        queue: str = conf.get('operators', 'default_queue'),
+        weight_rule: str = DEFAULT_WEIGHT_RULE,
+        queue: str = DEFAULT_QUEUE,
         pool: Optional[str] = None,
         pool_slots: int = 1,
         sla: Optional[timedelta] = None,
@@ -846,6 +850,9 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
     def node_id(self):
         return self.task_id
 
+    def get_dag(self) -> "Optional[DAG]":
+        return self._dag
+
     @property  # type: ignore[override]
     def dag(self) -> 'DAG':  # type: ignore[override]
         """Returns the Operator's DAG if set, otherwise raises an error"""
@@ -879,14 +886,6 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
     def has_dag(self):
         """Returns True if the Operator has been assigned to a DAG."""
         return self._dag is not None
-
-    @property
-    def dag_id(self) -> str:
-        """Returns dag id if it has one or an adhoc + owner"""
-        if self.has_dag():
-            return self.dag.dag_id
-        else:
-            return 'adhoc_' + self.owner
 
     deps: Iterable[BaseTIDep] = frozenset(
         {
@@ -938,38 +937,6 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
             if hasattr(self, field):
                 arg = getattr(self, field)
                 XComArg.apply_upstream_relationship(self, arg)
-
-    @property
-    def priority_weight_total(self) -> int:
-        """
-        Total priority weight for the task. It might include all upstream or downstream tasks.
-        depending on the weight rule.
-
-          - WeightRule.ABSOLUTE - only own weight
-          - WeightRule.DOWNSTREAM - adds priority weight of all downstream tasks
-          - WeightRule.UPSTREAM - adds priority weight of all upstream tasks
-
-        """
-        if self.weight_rule == WeightRule.ABSOLUTE:
-            return self.priority_weight
-        elif self.weight_rule == WeightRule.DOWNSTREAM:
-            upstream = False
-        elif self.weight_rule == WeightRule.UPSTREAM:
-            upstream = True
-        else:
-            upstream = False
-
-        if not self._dag:
-            return self.priority_weight
-        from airflow.models.dag import DAG
-
-        dag: DAG = self._dag
-        return self.priority_weight + sum(
-            map(
-                lambda task_id: dag.task_dict[task_id].priority_weight,
-                self.get_flat_relative_ids(upstream=upstream),
-            )
-        )
 
     @cached_property
     def operator_extra_link_dict(self) -> Dict[str, Any]:
@@ -1164,45 +1131,6 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
 
             self._do_render_template_fields(content, nested_template_fields, context, jinja_env, seen_oids)
 
-    def get_template_env(self) -> jinja2.Environment:
-        """Fetch a Jinja template environment from the DAG or instantiate empty environment if no DAG."""
-        return (
-            self.dag.get_template_env()
-            if self.has_dag()
-            else airflow.templates.SandboxedEnvironment(cache_size=0)
-        )
-
-    def prepare_template(self) -> None:
-        """
-        Hook that is triggered after the templated fields get replaced
-        by their content. If you need your operator to alter the
-        content of the file before the template is rendered,
-        it should override this method to do so.
-        """
-
-    def resolve_template_files(self) -> None:
-        """Getting the content of files for template_field / template_ext"""
-        if self.template_ext:
-            for field in self.template_fields:
-                content = getattr(self, field, None)
-                if content is None:
-                    continue
-                elif isinstance(content, str) and any(content.endswith(ext) for ext in self.template_ext):
-                    env = self.get_template_env()
-                    try:
-                        setattr(self, field, env.loader.get_source(env, content)[0])  # type: ignore
-                    except Exception as e:
-                        self.log.exception(e)
-                elif isinstance(content, list):
-                    env = self.dag.get_template_env()
-                    for i, item in enumerate(content):
-                        if isinstance(item, str) and any(item.endswith(ext) for ext in self.template_ext):
-                            try:
-                                content[i] = env.loader.get_source(env, item)[0]  # type: ignore
-                            except Exception as e:
-                                self.log.exception(e)
-        self.prepare_template()
-
     @provide_session
     def clear(
         self,
@@ -1259,27 +1187,6 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
             .order_by(TaskInstance.execution_date)
             .all()
         )
-
-    def get_flat_relative_ids(
-        self,
-        upstream: bool = False,
-        found_descendants: Optional[Set[str]] = None,
-    ) -> Set[str]:
-        """Get a flat set of relatives' ids, either upstream or downstream."""
-        if not self._dag:
-            return set()
-
-        if not found_descendants:
-            found_descendants = set()
-        relative_ids = self.get_direct_relative_ids(upstream)
-
-        for relative_id in relative_ids:
-            if relative_id not in found_descendants:
-                found_descendants.add(relative_id)
-                relative_task = self._dag.task_dict[relative_id]
-                relative_task.get_flat_relative_ids(upstream, found_descendants)
-
-        return found_descendants
 
     def get_flat_relatives(self, upstream: bool = False):
         """Get a flat list of relatives, either upstream or downstream."""
@@ -1355,16 +1262,6 @@ class BaseOperator(Operator, LoggingMixin, DAGNode, metaclass=BaseOperatorMeta):
             if content and isinstance(content, str):
                 self.log.info('Rendering template for %s', field)
                 self.log.info(content)
-
-    def get_direct_relative_ids(self, upstream: bool = False) -> Set[str]:
-        """
-        Get set of the direct relative ids to the current task, upstream or
-        downstream.
-        """
-        if upstream:
-            return self.upstream_task_ids
-        else:
-            return self.downstream_task_ids
 
     def get_direct_relatives(self, upstream: bool = False) -> Iterable["DAGNode"]:
         """
@@ -1635,7 +1532,7 @@ def _validate_kwarg_names_for_mapping(
 
 
 @attr.define(kw_only=True)
-class MappedOperator(DAGNode):
+class MappedOperator(Operator, LoggingMixin, DAGNode):
     """Object representing a mapped operator in a DAG"""
 
     def __repr__(self) -> str:
@@ -1668,8 +1565,11 @@ class MappedOperator(DAGNode):
 
     deps: Iterable[BaseTIDep] = attr.ib()
     operator_extra_links: Iterable['BaseOperatorLink'] = ()
-    params: Union[ParamsDict, dict] = attr.ib(factory=ParamsDict)
-    template_fields: Iterable[str] = attr.ib()
+    template_fields: Collection[str] = attr.ib()
+    template_ext: Collection[str] = attr.ib()
+
+    weight_rule: str = attr.ib()
+    priority_weight: int = attr.ib()
 
     subdag: None = attr.ib(init=False)
 
@@ -1682,6 +1582,22 @@ class MappedOperator(DAGNode):
     @deps.default
     def _deps_from_class(self):
         return self.operator_class.deps
+
+    @template_fields.default
+    def _template_fields_from_class(self):
+        return self.operator_class.template_fields
+
+    @template_ext.default
+    def _template_ext_from_class(self):
+        return self.operator_class.template_ext
+
+    @weight_rule.default
+    def _weight_rule_from_kwargs(self) -> str:
+        return self.partial_kwargs.get("weight_rule", DEFAULT_WEIGHT_RULE)
+
+    @priority_weight.default
+    def _priority_weight_from_kwargs(self) -> int:
+        return self.partial_kwargs.get("priority_weight", 1)
 
     @classmethod
     def from_operator(cls, operator: BaseOperator, mapped_kwargs: Dict[str, Any]) -> "MappedOperator":
@@ -1704,7 +1620,6 @@ class MappedOperator(DAGNode):
             owner=operator.owner,
             max_active_tis_per_dag=operator.max_active_tis_per_dag,
             deps=operator.deps,
-            params=operator.params,
         )
 
     @classmethod
@@ -1763,8 +1678,11 @@ class MappedOperator(DAGNode):
     def _template_fields_default(self):
         return self.operator_class.template_fields
 
+    def get_dag(self) -> "Optional[DAG]":
+        return self.dag
+
     @property
-    def node_id(self):
+    def node_id(self) -> str:
         return self.task_id
 
     def map(self, **kwargs) -> "MappedOperator":
@@ -1828,6 +1746,34 @@ class MappedOperator(DAGNode):
                 | {'template_fields'}
             )
         return cls.__serialized_fields
+
+    @property
+    def params(self) -> Union[dict, ParamsDict]:
+        return self.partial_kwargs.get("params", ParamsDict())
+
+    @property
+    def queue(self) -> str:
+        return self.partial_kwargs.get("queue", DEFAULT_QUEUE)
+
+    @property
+    def run_as_user(self) -> Optional[str]:
+        return self.partial_kwargs.get("run_as_user")
+
+    @property
+    def pool(self) -> Optional[str]:
+        return self.partial_kwargs.get("pool")
+
+    @property
+    def pool_slots(self) -> int:
+        return self.partial_kwargs.get("pool_slots", 1)
+
+    @property
+    def retries(self) -> Optional[int]:
+        return self.partial_kwargs.get("retries", DEFAULT_RETRIES)
+
+    @property
+    def executor_config(self) -> Optional[dict]:
+        return self.partial_kwargs.get("executor_config")
 
 
 # TODO: Deprecate for Airflow 3.0

--- a/tests/models/test_baseoperator.py
+++ b/tests/models/test_baseoperator.py
@@ -296,7 +296,7 @@ class TestBaseOperator:
         with pytest.raises(jinja2.exceptions.TemplateSyntaxError):
             task.render_template("{{ invalid expression }}", {})
 
-    @mock.patch("airflow.templates.SandboxedEnvironment", autospec=True)
+    @mock.patch("airflow.models.base.SandboxedEnvironment", autospec=True)
     def test_jinja_env_creation(self, mock_jinja_env):
         """Verify if a Jinja environment is created only once when templating."""
         task = MockOperator(task_id="op1", arg1="{{ foo }}", arg2="{{ bar }}")

--- a/tests/models/test_taskinstance.py
+++ b/tests/models/test_taskinstance.py
@@ -2253,13 +2253,6 @@ class TestTaskInstanceRecordTaskMapXComPush:
         with create_session() as session:
             session.query(TaskMap).delete()
 
-    def _run_ti_with_faked_mapped_dependants(self, ti):
-        # TODO: We can't actually put a MappedOperator in a DAG yet due to it
-        # lacking some functions we expect from BaseOperator, so we mock this
-        # instead to test what effect it has to TaskMap recording.
-        with mock.patch.object(ti.task, "has_mapped_dependants", new=lambda: True):
-            ti.run()
-
     @pytest.mark.parametrize("xcom_value", [[1, 2, 3], {"a": 1, "b": 2}, "abc"])
     def test_not_recorded_for_unused(self, dag_maker, xcom_value):
         """A value not used for task-mapping should not be recorded."""
@@ -2284,11 +2277,15 @@ class TestTaskInstanceRecordTaskMapXComPush:
             def push_something():
                 return "abc"
 
-            push_something()
+            @dag.task()
+            def pull_something(value):
+                print(value)
+
+            pull_something.map(value=push_something())
 
         ti = dag_maker.create_dagrun().task_instances[0]
         with pytest.raises(UnmappableXComPushed) as ctx:
-            self._run_ti_with_faked_mapped_dependants(ti)
+            ti.run()
 
         assert dag_maker.session.query(TaskMap).count() == 0
         assert ti.state == TaskInstanceState.FAILED
@@ -2309,11 +2306,15 @@ class TestTaskInstanceRecordTaskMapXComPush:
             def push_something():
                 return xcom_value
 
-            push_something()
+            @dag.task()
+            def pull_something(value):
+                print(value)
+
+            pull_something.map(value=push_something())
 
         dag_run = dag_maker.create_dagrun()
         ti = next(ti for ti in dag_run.task_instances if ti.task_id == "push_something")
-        self._run_ti_with_faked_mapped_dependants(ti)
+        ti.run()
 
         task_map = dag_maker.session.query(TaskMap).one()
         assert task_map.dag_id == "test_written_task_map"

